gap/prop.gi

```diff
@@ -555,3 +555,108 @@ function(gr)
   return true;
 
 end);
+
+#
+
+InstallMethod(IsHamiltonianDigraph, "for a digraph",
+[IsDigraph],
+function(g)
+  local g2, out, Indegs, Outdegs, Degrees, n, check,
+  dominatedcheck, dominatingordominatedcheck, i, j, k, dp;
+
+  if DigraphNrVertices(g) <= 1 and IsEmptyDigraph(g) then
+     return true;
+  fi;
+  if not IsStronglyConnectedDigraph(g) then
+     return false;
+  fi;
+  g2 := DigraphRemoveLoops(g);
```

Comment:
This is unnecessarily wasteful, since the digraph `g` is copied here even if it has no loops anyway. Better do `if DigraphHasLoops(g) then ...` and otherwise set `g2 := g`